pkg/pillar/types/global.go

```diff
@@ -325,6 +329,35 @@ const (
 	ConfigItemTypeTriState
 )
 
+var (
+	// SyslogKernelLogLevelStr is a string representation of syslog/kernel
+	// loglevels.
+	SyslogKernelLogLevelStr = [8]string{
+		"emerg", "alert", "crit", "err", "warning", "notice", "info", "debug",
+	}
+	// SyslogKernelLogLevelNum is a number representation of syslog/kernel
+	// loglevels. Plus compatibility with logrus is kept. Be aware, logrus
+	// numbers are not mapped 1 to 1.
+	SyslogKernelLogLevelNum = map[string]uint32{
+		"emerg":    0,
+		"panic":    0, // logrus
+		"alert":    1,
+		"crit":     2,
+		"critical": 2,
+		"fatal":    2, // logrus
+		"err":      3,
+		"error":    3,
+		"warning":  4,
+		"warn":     4, // logrus
+		"notice":   5,
+		"info":     6,
+		"debug":    7,
+		"trace":    7, // logrus
```

Comment:
debug and trace are two different levels in logrus, yet you map them to the same number.

Should info and notice be mapped to the same number instead?

Reply:
Syslog has 7 loglevels, logrus 6. I can't map them 1 to 1. I do not think "trace" mapped to "debug" is really important, since syslog/kernel does not have "trace" and we use it only for this setting. Logrus is kept untouched.

"info" and "notice" is not logrus, this is original syslog/kernel log levels.

Reply:
I still don't understand the compatibility then. 

It might be easier to not have such a thing and instead list the allowed strings for the existing and new level fields in CONFIG-PROPERTIES.md

Reply:
The intention is the following:

 zcli edge-node update $NODE --config="debug.default.loglevel:fatal" 
 
^^ this exists for ages and probably get used to use "fatal" or "panic"
  
 zcli edge-node update $NODE --config="debug.kernel.loglevel:critical"
 
 ^^ this is introduced by this PR and now user has to specify different log levels, which kernel and syslog have, which can be confusing and error prone. (or not? this is the question.)
 
So only for new settings "debug.kernel.loglevel" and "debug.syslog.loglevel" I kept the compatibility with logrus intentionally to avoid any possible mistakes. Mapping is not 1 to 1 which is not so critical.

@eriknordmark @milan-zededa what do you think? Can remove this compatibility if you promise me not to confuse these two log levels naming :)

Reply:
I'm OK if we drop the compatibility since you added links for the set of allowed values in CONFIG-PROPERTIES.md

Reply:
Will remove and update CONFIG-PROPERTIES.md then. 

If you specify wrong level which cannot be parsed you will see the error something like logrus log levels can't be parsed, return to the default "info". (do not remember exactly).